src/main/java/org/commcare/suite/model/MenuLoader.java

```diff
@@ -145,12 +140,48 @@ private boolean menuIsRelevant(SessionWrapperInterface sessionWrapper, Menu m) t
         return true;
     }
 
+    public boolean menuAssertionsPass(SessionWrapperInterface sessionWrapper, Menu m) throws XPathSyntaxException {
+        AssertionSet assertions = m.getAssertions();
+        Vector<String> assertionXPathStrings = assertions.getAssertionsXPaths();
+        if (!assertionXPathStrings.isEmpty()) {
```

Comment:
should not be necessary as the for loop inside should exit if the list is empty

Reply:
fixed here https://github.com/dimagi/commcare-core/pull/1267/commits/1d166f9270511ad1fac11eb7eac503bccc87145c